### ISSUE TITLE
Add ability to generate prefilled letters

### ISSUE THREAD

**Issue by eglipeter** [nextcloud/contacts#636]
A feature which I am missing is that one can download an office document file for a new letter where the address of a contact is already filled in. Therefore, it should be possible to upload a template document for each contact group. When initiating the download, the app should replace placeholders in the document with the values of corresponding fields.

One might think even further and also allow the user to upload an already written letter with placeholders in order to create easily bulk letters for selected contacts.

As formats, I think the ODT format should be supported and maybe also DOC/DOCX formats.

Yours, 

Peter

**Comment by skjnldsv** [comment]
Hi peter!
I'm having a bit of trouble understanding what you suggest.
I don't see the link between the contacts and the documents, could you explain a bit further? :)

Thanks!

**Comment by eglipeter** [comment]
My idea is that a user can upload an office document (porobably somewhere in the settings) where placeholders like e.g. `##name##`,`##street##`,`##city##` etc are in. These placeholders could be used especially in the address field of a letter. Another possible field could be `##date##` where the current date is inserted (you dont't want to use the automatic date fields in your office suite when you save the file for archiving). Then, if the user wants to write a new letter to one of his contacts, he can simply download an office document where the correct address is already filled in. He then just needs to enter the text of his letter. By that you avoid typos and speed up the process of writing new letters. This would be part one. For that, a new button would be shown on each contact detail page which allows to download such a prefilled letter where the address and the saluation are filled in correctly for this specific contact. This is (in my case) the more important part of the proposal.
It would be nice if the user could upload different templates for different contact groups (e.g. for private and business contacts where the letterhead could have two different return addresses). The letter text of such a template would be empty.

The other idea is that a user should be able to upload an office document where the letter text is already filled for the generation of mass letters. Think of the case, you want to inform all your customers about the new GDPR with classical (physical) mail. Then, it would be nice if you could create a letter where you insert some placeholders for the address etc. Then, you upload the letter to Nextcloud and Nextcloud generates copies of this letter and replaces the "search strings" by the correct fields of the contact database. The user obtains then a folder in the Files section or a ZIP download where all filled copies of the letter are in. 

To sum up, although the technical implementation of these two features are closely related, the purpose of them are rather different. In the former case the goal is to write a letter to a single customer and you are only "lazy" to type again (or use Ctrl+C) the address of a specific case. In the later case, you want to mail out the same letter to a large number of customers.

I hope it got a little clearer what I mean. If not, don't hessitate to ask again.

Yours,
Peter

**Comment by skjnldsv** [comment]
Hey @eglipeter!
The contacts app is not really the place for this suggestion! :)
If you're thinking about using onlyoffice or collabora on nextcloud for that, please create an issue on those repositories !
This is a very interesting suggestion! 
https://github.com/nextcloud/richdocuments
https://github.com/ONLYOFFICE/onlyoffice-nextcloud

I'll close this issue then. Your last message was very clear, I suggest you use copy it on the next issue :)
